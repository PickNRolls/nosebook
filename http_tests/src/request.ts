import supertest, { Test } from 'supertest';
import TestAgent from 'supertest/lib/agent';

const HOST = 'http://backend:8080';
const SESSION_HEADER_KEY = 'X-Auth-Session-Id';

class TestAgentWrapper {
  private host: string;
  private agent: TestAgent;

  public constructor(host: string) {
    this.host = host;
    this.agent = supertest(host);
  }

  public unwrap(): TestAgent {
    return this.agent;
  }

  public extend(prefixUrl: string): TestAgentWrapper {
    return new TestAgentWrapper(this.host + prefixUrl);
  }

  private hook<M extends 'get' | 'post'>(method: M, url: string): Test {
    const test = this.agent[method](url);

    test.set(SESSION_HEADER_KEY, 'bb23af03-be50-4bce-b729-b259b2e02e54');

    return test;
  }

  public get(url: string): Test {
    return this.hook('get', url);
  }

  public post(url: string): Test {
    return this.hook('post', url);
  }
};

export default new TestAgentWrapper(HOST);


