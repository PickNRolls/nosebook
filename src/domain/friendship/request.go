package domainfriendship

import (
	"nosebook/src/errors"
	"time"

	"github.com/google/uuid"
)

type FriendRequest struct {
	RequesterId uuid.UUID `json:"requesterId" db:"requester_id"`
	ResponderId uuid.UUID `json:"responderId" db:"responder_id"`
	Message     string    `json:"message" db:"message"`
	Accepted    bool      `json:"accepted" db:"accepted"`
	Viewed      bool      `json:"viewed" db:"viewed"`
	CreatedAt   time.Time `json:"createdAt" db:"created_at"`

	events []Event
}

func New(
	requesterId uuid.UUID,
	responderId uuid.UUID,
	message string,
	accepted bool,
	viewed bool,
	createdAt time.Time,
	raiseCreatedEvent bool,
) *FriendRequest {
	output := &FriendRequest{
		RequesterId: requesterId,
		ResponderId: responderId,
		Message:     message,
		Accepted:    accepted,
		Viewed:      viewed,
		CreatedAt:   createdAt,
		events:      make([]Event, 0),
	}

	if raiseCreatedEvent {
		output.raiseEvent(NewCreatedEvent())
	}

	return output
}

func (this *FriendRequest) raiseEvent(event Event) {
	this.events = append(this.events, event)
}

func (this *FriendRequest) Events() []Event {
	return this.events
}

func (this *FriendRequest) ViewBy(userId uuid.UUID) *errors.Error {
	if userId != this.ResponderId {
		return errors.New("Friendship Error", "Только получатель может отметить заявку просмотренной")
	}

	this.Viewed = true
	this.raiseEvent(NewViewedEvent())

	return nil
}

func (this *FriendRequest) AcceptBy(userId uuid.UUID) *errors.Error {
	if userId != this.ResponderId {
		return errors.New("Friendship Error", "Только получатель может принять заявку")
	}

	this.Accepted = true
	this.Viewed = true
	this.raiseEvent(NewAcceptedEvent())

	return nil
}

func (this *FriendRequest) DenyBy(userId uuid.UUID) *errors.Error {
	if userId != this.ResponderId {
		return errors.New("Friendship Error", "Только получатель может отклонить заявку")
	}

	this.Accepted = false
	this.Viewed = true
	this.raiseEvent(NewDeniedEvent())

	return nil
}

func (this *FriendRequest) RemoveBy(userId uuid.UUID) *errors.Error {
	if !this.Accepted {
		return errors.New("Friendship Error", "Нельзя удалить непринятую заявку")
	}

	this.Accepted = false
	previousRequesterId := this.RequesterId
	previousResponderId := this.ResponderId

	if this.RequesterId == userId {
		this.RequesterId, this.ResponderId = this.ResponderId, this.RequesterId
	}

	this.raiseEvent(NewRemovedEvent(previousRequesterId, previousResponderId))

	return nil
}
