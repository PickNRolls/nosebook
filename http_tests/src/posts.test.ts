import supertest from 'supertest';
import request from './request';

describe('posts', () => {
  let posts = request.extend('/posts');
  let createdPostId: string;

  test('GET /', async () => {
    let lastResponse = await posts
      .get('/')
      .query({
        ownerId: '1ae02f69-ea1a-4308-b825-0e5896e652e4',
        authorId: '1ae02f69-ea1a-4308-b825-0e5896e652e4',
      })
      .expect(200)

    expect(lastResponse.body).toMatchSnapshot();

    lastResponse = await posts
      .get('/')
      .query({
        ownerId: '1ae02f69-ea1a-4308-b825-0e5896e652e4',
        authorId: '1ae02f69-ea1a-4308-b825-0e5896e652e4',
        cursor: lastResponse.body.data.next,
      })
      .expect(200);

    expect(lastResponse.body).toMatchSnapshot();
  });

  test('POST /publish', async () => {
    let response = await posts
      .post('/publish')
      .send({
        ownerId: '1ae02f69-ea1a-4308-b825-0e5896e652e4',
        message: 'my test message',
      })
      .expect(200);

    expect(response.body).toEqual(expect.objectContaining({
      data: {
        id: expect.any(String),
      },
      ok: true,
    }));
    createdPostId = response.body.data.id;
  });

  test('POST /remove', async () => {
    const response = await posts
      .post('/remove')
      .send({
        id: createdPostId,
      }).expect(200);

    expect(response.body).toEqual(expect.objectContaining({
      data: {
        id: createdPostId,
      },
      ok: true,
    }));
  });
});

