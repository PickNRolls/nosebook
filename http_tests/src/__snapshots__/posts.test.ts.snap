// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`posts GET / 1`] = `
{
  "data": {
    "data": [
      {
        "author": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "createdAt": "2024-02-16T14:36:59Z",
        "id": "c7b7bf17-38f9-4ed5-b0a8-501a90f7c8e2",
        "likes": {
          "count": 0,
          "liked": false,
          "randomFiveLikers": [],
        },
        "message": "post message 5",
        "owner": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "permissions": {
          "remove": false,
          "update": false,
        },
        "recentComments": {
          "data": [],
          "totalCount": 0,
        },
      },
      {
        "author": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "createdAt": "2024-02-16T14:36:59Z",
        "id": "c7b7bf17-38f9-4ed5-b0a8-011a90f7c8e2",
        "likes": {
          "count": 0,
          "liked": false,
          "randomFiveLikers": [],
        },
        "message": "post message 11",
        "owner": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "permissions": {
          "remove": false,
          "update": false,
        },
        "recentComments": {
          "data": [],
          "totalCount": 0,
        },
      },
      {
        "author": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "createdAt": "2024-02-16T14:36:59Z",
        "id": "c7b7bf17-38f9-4ed5-b0a8-001a90f7c8e2",
        "likes": {
          "count": 0,
          "liked": false,
          "randomFiveLikers": [],
        },
        "message": "post message 10",
        "owner": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "permissions": {
          "remove": false,
          "update": false,
        },
        "recentComments": {
          "data": [],
          "totalCount": 0,
        },
      },
      {
        "author": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "createdAt": "2024-02-16T14:36:57Z",
        "id": "c7b7bf17-38f9-4ed5-b0a8-901a90f7c8e0",
        "likes": {
          "count": 0,
          "liked": false,
          "randomFiveLikers": [],
        },
        "message": "post message 9",
        "owner": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "permissions": {
          "remove": false,
          "update": false,
        },
        "recentComments": {
          "data": [],
          "totalCount": 0,
        },
      },
      {
        "author": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "createdAt": "2024-02-16T14:36:57Z",
        "id": "c7b7bf17-38f9-4ed5-b0a8-501a90f7c8e0",
        "likes": {
          "count": 0,
          "liked": false,
          "randomFiveLikers": [],
        },
        "message": "post message 4",
        "owner": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "permissions": {
          "remove": false,
          "update": false,
        },
        "recentComments": {
          "data": [],
          "totalCount": 0,
        },
      },
      {
        "author": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "createdAt": "2024-02-16T14:36:53Z",
        "id": "c7b7bf17-38f9-4ed5-b0a8-801a90f7c8e9",
        "likes": {
          "count": 0,
          "liked": false,
          "randomFiveLikers": [],
        },
        "message": "post message 8",
        "owner": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "permissions": {
          "remove": false,
          "update": false,
        },
        "recentComments": {
          "data": [],
          "totalCount": 0,
        },
      },
      {
        "author": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "createdAt": "2024-02-16T14:36:53Z",
        "id": "c7b7bf17-38f9-4ed5-b0a8-501a90f7c8e9",
        "likes": {
          "count": 0,
          "liked": false,
          "randomFiveLikers": [],
        },
        "message": "post message 3",
        "owner": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "permissions": {
          "remove": false,
          "update": false,
        },
        "recentComments": {
          "data": [],
          "totalCount": 0,
        },
      },
      {
        "author": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "createdAt": "2024-02-16T14:36:48Z",
        "id": "c7b7bf17-38f9-4ed5-b0a8-701a90f7c8e8",
        "likes": {
          "count": 0,
          "liked": false,
          "randomFiveLikers": [],
        },
        "message": "post message 7",
        "owner": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "permissions": {
          "remove": false,
          "update": false,
        },
        "recentComments": {
          "data": [],
          "totalCount": 0,
        },
      },
      {
        "author": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "createdAt": "2024-02-16T14:36:48Z",
        "id": "c7b7bf17-38f9-4ed5-b0a8-501a90f7c8e8",
        "likes": {
          "count": 2,
          "liked": false,
          "randomFiveLikers": [
            {
              "firstName": "Ilya",
              "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
              "lastName": "Blinkov",
              "lastOnlineAt": "2024-10-10T10:01:02Z",
              "nick": "drugtester",
              "online": false,
            },
            {
              "firstName": "Marina",
              "id": "48683858-796c-45ad-a361-9e3d6d003354",
              "lastName": "Graf",
              "lastOnlineAt": "2024-10-10T10:10:02Z",
              "nick": "mmm",
              "online": true,
            },
          ],
        },
        "message": "post message 2",
        "owner": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "permissions": {
          "remove": false,
          "update": false,
        },
        "recentComments": {
          "data": [],
          "totalCount": 0,
        },
      },
      {
        "author": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "createdAt": "2024-02-16T14:36:38Z",
        "id": "c7b7bf17-38f9-4ed5-b0a8-601a90f7c8e7",
        "likes": {
          "count": 0,
          "liked": false,
          "randomFiveLikers": [],
        },
        "message": "post message 6",
        "owner": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "permissions": {
          "remove": false,
          "update": false,
        },
        "recentComments": {
          "data": [],
          "totalCount": 0,
        },
      },
    ],
    "next": "c7b7bf17-38f9-4ed5-b0a8-601a90f7c8e7/2024-02-16T14:36:38Z",
    "totalCount": 11,
  },
  "ok": true,
}
`;

exports[`posts GET / 2`] = `
{
  "data": {
    "data": [
      {
        "author": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "createdAt": "2024-02-16T14:36:38Z",
        "id": "c7b7bf17-38f9-4ed5-b0a8-501a90f7c8e7",
        "likes": {
          "count": 7,
          "liked": false,
          "randomFiveLikers": [
            {
              "firstName": "Ilya",
              "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
              "lastName": "Blinkov",
              "lastOnlineAt": "2024-10-10T10:01:02Z",
              "nick": "drugtester",
              "online": false,
            },
            {
              "firstName": "Marina",
              "id": "48683858-796c-45ad-a361-9e3d6d003354",
              "lastName": "Graf",
              "lastOnlineAt": "2024-10-10T10:10:02Z",
              "nick": "mmm",
              "online": true,
            },
            {
              "firstName": "Sasha",
              "id": "baa0e8bc-385f-4314-9580-29855aff2229",
              "lastName": "Provodnikov",
              "lastOnlineAt": "2024-10-10T10:10:02Z",
              "nick": "yyy",
              "online": true,
            },
            {
              "firstName": "Tolber",
              "id": "37d28fdf-99bc-44b5-8df9-6a3b1a36f177",
              "lastName": "Ovcharenko",
              "lastOnlineAt": "2024-10-10T10:10:02Z",
              "nick": "tolber01",
              "online": true,
            },
            {
              "firstName": "Sanal",
              "id": "2db640fd-7aa4-4bba-8ee6-3935b700297a",
              "lastName": "Mandjiev",
              "lastOnlineAt": "2024-10-10T10:10:02Z",
              "nick": "sanal",
              "online": true,
            },
          ],
        },
        "message": "post message",
        "owner": {
          "firstName": "Ilya",
          "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
          "lastName": "Blinkov",
          "lastOnlineAt": "2024-10-10T10:01:02Z",
          "nick": "drugtester",
          "online": false,
        },
        "permissions": {
          "remove": false,
          "update": false,
        },
        "recentComments": {
          "data": [
            {
              "author": {
                "firstName": "Test",
                "id": "ed1a3fd0-4d0b-4961-b4cd-cf212357740d",
                "lastName": "Tester",
                "lastOnlineAt": "2024-10-10T10:10:10Z",
                "nick": "test_tester",
                "online": true,
              },
              "createdAt": "2024-02-16T15:36:38Z",
              "id": "120c79b8-3927-48b7-a308-1ffd3db6036f",
              "likes": {
                "count": 0,
                "liked": false,
                "randomFiveLikers": [],
              },
              "message": "comment for permissions test",
              "permissions": {
                "remove": true,
                "update": true,
              },
            },
            {
              "author": {
                "firstName": "Ilya",
                "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
                "lastName": "Blinkov",
                "lastOnlineAt": "2024-10-10T10:01:02Z",
                "nick": "drugtester",
                "online": false,
              },
              "createdAt": "2024-02-16T15:36:55Z",
              "id": "d0023f4d-8d7f-4907-9438-d2ed2a9661f0",
              "likes": {
                "count": 0,
                "liked": false,
                "randomFiveLikers": [],
              },
              "message": "3nd comment message",
              "permissions": {
                "remove": false,
                "update": false,
              },
            },
            {
              "author": {
                "firstName": "Ilya",
                "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
                "lastName": "Blinkov",
                "lastOnlineAt": "2024-10-10T10:01:02Z",
                "nick": "drugtester",
                "online": false,
              },
              "createdAt": "2024-02-16T15:37:55Z",
              "id": "d0023f4d-8d7f-4907-9438-d2ed2a9661f1",
              "likes": {
                "count": 0,
                "liked": false,
                "randomFiveLikers": [],
              },
              "message": "4nd comment message",
              "permissions": {
                "remove": false,
                "update": false,
              },
            },
            {
              "author": {
                "firstName": "Ilya",
                "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
                "lastName": "Blinkov",
                "lastOnlineAt": "2024-10-10T10:01:02Z",
                "nick": "drugtester",
                "online": false,
              },
              "createdAt": "2024-02-16T15:38:55Z",
              "id": "d0023f4d-8d7f-4907-9438-d2ed2a9661f2",
              "likes": {
                "count": 0,
                "liked": false,
                "randomFiveLikers": [],
              },
              "message": "5nd comment message",
              "permissions": {
                "remove": false,
                "update": false,
              },
            },
            {
              "author": {
                "firstName": "Ilya",
                "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
                "lastName": "Blinkov",
                "lastOnlineAt": "2024-10-10T10:01:02Z",
                "nick": "drugtester",
                "online": false,
              },
              "createdAt": "2024-02-16T15:39:55Z",
              "id": "d0023f4d-8d7f-4907-9438-d2ed2a9661f3",
              "likes": {
                "count": 0,
                "liked": false,
                "randomFiveLikers": [],
              },
              "message": "6nd comment message",
              "permissions": {
                "remove": false,
                "update": false,
              },
            },
          ],
          "next": "d0023f4d-8d7f-4907-9438-d2ed2a9661f3/2024-02-16T15:39:55Z",
          "totalCount": 8,
        },
      },
    ],
    "totalCount": 11,
  },
  "ok": true,
}
`;

exports[`posts permissions GET /:id, has no permissions 1`] = `
{
  "data": {
    "author": {
      "firstName": "Ilya",
      "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
      "lastName": "Blinkov",
      "lastOnlineAt": "2024-10-10T10:01:02Z",
      "nick": "drugtester",
      "online": false,
    },
    "createdAt": "2024-02-16T14:36:59Z",
    "id": "c7b7bf17-38f9-4ed5-b0a8-011a90f7c8e2",
    "likes": {
      "count": 0,
      "liked": false,
      "randomFiveLikers": [],
    },
    "message": "post message 11",
    "owner": {
      "firstName": "Ilya",
      "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
      "lastName": "Blinkov",
      "lastOnlineAt": "2024-10-10T10:01:02Z",
      "nick": "drugtester",
      "online": false,
    },
    "permissions": {
      "remove": false,
      "update": false,
    },
    "recentComments": {
      "data": [],
      "totalCount": 0,
    },
  },
  "ok": true,
}
`;

exports[`posts permissions GET /:id, has permissions auth.user == post.author 1`] = `
{
  "data": {
    "author": {
      "firstName": "Test",
      "id": "ed1a3fd0-4d0b-4961-b4cd-cf212357740d",
      "lastName": "Tester",
      "lastOnlineAt": "2024-10-10T10:10:10Z",
      "nick": "test_tester",
      "online": true,
    },
    "createdAt": "2024-02-16T14:36:38Z",
    "id": "27b7bf17-38f9-4ed5-b0a8-501a90f7c8e7",
    "likes": {
      "count": 0,
      "liked": false,
      "randomFiveLikers": [],
    },
    "message": "test permissions post with only author id",
    "owner": {
      "firstName": "Ilya",
      "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
      "lastName": "Blinkov",
      "lastOnlineAt": "2024-10-10T10:01:02Z",
      "nick": "drugtester",
      "online": false,
    },
    "permissions": {
      "remove": true,
      "update": true,
    },
    "recentComments": {
      "data": [],
      "totalCount": 0,
    },
  },
  "ok": true,
}
`;

exports[`posts permissions GET /:id, has permissions auth.user == post.owner 1`] = `
{
  "data": {
    "author": {
      "firstName": "Ilya",
      "id": "1ae02f69-ea1a-4308-b825-0e5896e652e4",
      "lastName": "Blinkov",
      "lastOnlineAt": "2024-10-10T10:01:02Z",
      "nick": "drugtester",
      "online": false,
    },
    "createdAt": "2024-02-16T14:36:38Z",
    "id": "27b7bf27-38f9-4ed5-b0a8-501a90f7c8e7",
    "likes": {
      "count": 0,
      "liked": false,
      "randomFiveLikers": [],
    },
    "message": "test permissions post with only owner id",
    "owner": {
      "firstName": "Test",
      "id": "ed1a3fd0-4d0b-4961-b4cd-cf212357740d",
      "lastName": "Tester",
      "lastOnlineAt": "2024-10-10T10:10:10Z",
      "nick": "test_tester",
      "online": true,
    },
    "permissions": {
      "remove": true,
      "update": false,
    },
    "recentComments": {
      "data": [],
      "totalCount": 0,
    },
  },
  "ok": true,
}
`;
